src/sync_endpoint.rs

```diff
@@ -0,0 +1,35 @@
+//! Endpoint -> Famedly Zitadel sync tool.
+
+use anyhow::{bail, Result};
+
+use crate::{
+	config::{EndpointConfig, FeatureFlags},
+	zitadel::Zitadel,
+	Config, SyncSource,
+};
+
+/// Endpoint sync source
+pub struct EndpointSync {
+	/// Endpoint configuration
+	pub(crate) endpoint_config: EndpointConfig,
+	/// Optional set of features
+	pub(crate) feature_flags: FeatureFlags,
+}
+
+impl SyncSource for EndpointSync {
+	fn new(config: &Config) -> Result<Self> {
+		Ok(Self {
+			endpoint_config: match config.endpoint.clone() {
+				Some(endpoint) => endpoint,
+				None => bail!("Endpoint configuration is missing"),
+			},
+			feature_flags: config.feature_flags.clone(),
+		})
+	}
+
+	async fn sync(&self, zitadel_client: &Zitadel) -> Result<()> {
+		todo!("Fetch the email from the endpoint");
+		let emails = vec![];
+		zitadel_client.delete_users_by_email(emails).await?;
```

Comment:
I think the sync call should be in the `lib.rs` crate, and as discussed in `zitadel.rs`, we probably should not have divergent implementations for how each source syncs its users.